intake/catalog/tests/test_local.py

```diff
@@ -121,19 +121,19 @@ def test_use_source_plugin_from_config(catalog1):
 
 
 def test_get_dir():
-    assert get_dir("file:///path/catalog.yml") == "file:///path"
-    assert get_dir("https://example.com/catalog.yml") == "https://example.com"
+    assert get_dir("file:///path/catalog.yml") == "file://path/"
```

Comment:
I am still confused by `.replace(':/', '://')`, and its apparent inverse which happened here.

Does as_posix() silently change "file://"->"file:/" ? What then happens to paths that genuinely have colons, like "x:/path/..."?